confdgnmi/testtool/gNMI_Interface/Get.resource

```diff
@@ -46,3 +48,43 @@ Verify Get failure for unsupported encodings
 #     FOR  ${model}  IN  @{model_names}
 #         Verify Get of model  ${model}  ${pass-condition-keyword}
 #     END
+
+Iterate leaves "${leaves}"" of path "${path}"
+    [Documentation]    Retrieve the item from specified path, and verify that response
+    ...                includes some data for it.
+    ...                Cleanup the gNMI library state after to allow other clean iterations.
+    Should Be True    ${leaves} is not ${None}  Missing leaves to test for path "${path}"!
+    Should Not Be Empty    ${leaves}
+    When Verify Get of  ${path}
+    ${count}=  Get Last Updates Count
+    # Then Should Be Equal As Integers    ${count}  1
+    FOR  ${leaf}  IN  @{leaves}
+        Run Keyword And Continue On Failure  Then Check Last Updates Include    ${leaf}
+    END
+    [Teardown]    Teardown gNMI state
```

Comment:
IMHO it is wanted here is to iterate all the configured items independently... if one of the leaves was is-configured or not implemented by device, it's no reason not to test other ones as well...

We've seen (on different projects) a problem of user having to re-iterate test-runs or log mining repeatedly step by step, with small progress in between each "iteration" instead of getting the bulk information to allow batch fixing/updates...

Reply:
I believe you are talking about `Run Keyword And ...` - that is ok, I understand the reasons. What I meant is that the settings `[Teardown]` is not necessary, everything would work the same if it was not there, I think. (Yes, I was wrong in the part _no further iterations are run_; but the point remains.)

Reply:
it is, unfortunately... https://github.com/robotframework/robotframework/issues/2668